vulture_os/toolkit/api_parser/apex/apex.py

```diff
@@ -0,0 +1,199 @@
+        self.apex_spyware_timestamp = data.get("apex_spyware_timestamp")
+        self.apex_web_security_timestamp = data.get("apex_web_security_timestamp")
+        self.apex_security_timestamp = data.get("apex_security_timestamp")
+        self.apex_ncie_timestamp = data.get("apex_ncie_timestamp")
+        self.apex_cncdetection_timestamp = data.get("apex_cncdetection_timestamp")
+        self.apex_filehashdetection_timestamp = data.get("apex_filehashdetection_timestamp")
+        self.apex_Predictive_Machine_Learning_timestamp = data.get("apex_Predictive_Machine_Learning_timestamp")
+        self.apex_Sandbox_Detection_Log_timestamp = data.get("apex_Sandbox_Detection_Log_timestamp")
+        self.apex_EACV_Information_timestamp = data.get("apex_EACV_Information_timestamp")
+        self.apex_Managed_Product_Logged_Information_timestamp = data.get("apex_Managed_Product_Logged_Information_timestamp")
+        self.apex_Attack_Discovery_Detections_timestamp = data.get("apex_Attack_Discovery_Detections_timestamp")
+        self.apex_pattern_updated_status_timestamp = data.get("apex_pattern_updated_status_timestamp")
+        self.apex_engine_updated_status_timestamp = data.get("apex_engine_updated_status_timestamp")
+        self.apex_product_auditing_events_timestamp = data.get("apex_product_auditing_events_timestamp")
+        self.apex_intrusion_prevention_timestamp = data.get("apex_intrusion_prevention_timestamp")
+
+
+        self.session = None
+        self.HEADERS = {'Content-Type': 'application/json;charset=utf-8'}
+        self.canonicalRequestHeaders = ''
+        self.useRequestBody = ''
+
+    def create_checksum(self, http_method, raw_url, headers, request_body):
+        string_to_hash = http_method.upper() + '|' + raw_url.lower() + '|' + headers + '|' + request_body
+        base64_string = base64.b64encode(hashlib.sha256(str.encode(string_to_hash)).digest()).decode('utf-8')
+        return base64_string
+    
+    def create_jwt_token(self, application_id, api_key, http_method, raw_url, headers, request_body,
+        iat=timezone.now(), algorithm='HS256', version='V1'):
+        payload = {'appid': application_id,
+               'iat': iat,
+               'version': version,
+               'checksum': self.create_checksum(http_method, raw_url, headers, request_body)}
+        token = jwt.encode(payload, api_key, algorithm=algorithm)
+        return token
+
+    def _connect(self, raw_url):
+        try:
+            # New session because timeout is very short
+            self.session = requests.Session()
+            logger.error(f"raw_url : {raw_url}", extra={'frontend': str(self.frontend)})
+            jwt_token = self.create_jwt_token(self.apex_application_id, self.apex_api_key, 'GET',
+                              raw_url,
+                              self.canonicalRequestHeaders, self.useRequestBody, iat=timezone.now().timestamp())
```

Comment:
careful: datetime.timestamp() returns a float with millisecond precision, is that what is wanted?